lib/reek/smells/uncommunicative_variable_name.rb

```diff
@@ -24,13 +26,17 @@ class UncommunicativeVariableName < SmellDetector
       # The name of the config field that lists the regexps of
       # smelly names to be reported.
       REJECT_KEY = 'reject'.freeze
-      DEFAULT_REJECT_SET = [/^.$/, /[0-9]$/, /[A-Z]/].freeze
+      DEFAULT_REJECT_SET = [
+        /^[a-z]$/, # single-character names
+        /[0-9]$/,  # any name ending with a number
+        /[A-Z]/    # camelCaseVariableNames
+      ].freeze
 
       # The name of the config field that lists the specific names that are
       # to be treated as exceptions; these names will not be reported as
       # uncommunicative.
       ACCEPT_KEY = 'accept'.freeze
-      DEFAULT_ACCEPT_SET = ['_'].freeze
+      DEFAULT_ACCEPT_SET = [/^_/].freeze
```

Comment:
This makes us accept `_aBadName42`. I think the original intent was to accept `_` as the only valid single-character name.